### PR TITLE
Create a new Footer element and move Feedback link there

### LINKED ISSUE #48: De-duplicate the feedback form footer
https://github.com/SuffolkLITLab/docassemble-AssemblyLine/blob/c9607e1b5cfedda7e5877ef33bdabbffe64ad5bb/docassemble/AssemblyLine/data/questions/al_visual.yml#L49-L53

Both `if get_config('debug'):` and its `else:` seem to use the same code. If that's correct, we can get rid of the `if`.

### LINKED ISSUE #58: Add a "permalink" button, perhaps in a footer
We don't use the screen parts: footer element yet, but this might work there.

I think this could solve two problems: one is to let individual promote the tool to friends. The second is to allow outside entities to link to the form correctly. Right now people tend to copy/paste the URL from the URL bar. That leads to a /run/interview URL, which resumes the last place someone was in the form. That's not so ideal since there's no easy way to restart.

If we do this, we might also want a few other related tasks:

1. Share the link to your interview session
2. Send an email or text to tell someone about the tool
3. Restart/new session button that prompts you to create an account first if you are not already logged in.

### LINKED ISSUE #61: Figure out a way and place to add "freshness" labels to court forms
All forms should have a last-updated date somewhere.

### LINKED ISSUE #130: Display the version number somewhere on the production server. We need an "about" screen somewhere?

### LINKED ISSUE #137: Create a footer to house a link to "Feedback", "About", and "Share" modals/forms
At the 2021-05-04 Admin meeting, we agreed it was worth trying out a way to display a "share" button and an "about" button.

There was no consensus on the best place to put it. One thought was the hamburger/triangle menu, but the menu is not very discoverable. That is OK for an "about" page, but not share or feedback. The problem with putting it anywhere else is that the screen is already fairly busy.

The thing we agreed to try for now was:

Changing the "complete this survey" link to just say "Feedback", moving it to the footer, and adding a one-word "Share" and "About" link.

We are not sure if icons will be understandable or not. I think we could give some font-awesome icons a try. Here is what would be under each link:

1. **Feedback** would be the existing form feedback link.
2. **Share** would give you two options: a link to advertise the interview to someone with a similar need, and a link to share your current in-progress interview. This would ideally be a modal.
3. **About** would contain information about the CourtFormsOnline project, the last date the interview was updated, and perhaps a few other things.

Relates to #61, #58, #130 and perhaps other issues

### PR DESCRIPTION
**Pull request by nonprofittechy**
Fix #48, Fix #61, Fix #130, Fix #58, Fix #137

Creates a new footer element with basic "about" screen functionality, feedback link, and a "share this interview" functionality.

### CODE REVIEW

**Comment by nonprofittechy** [comment]
Feedback from Standup on 5/14:

* Move the names of contributors to the bottom on the about page
* use two sets of radio buttons on the "share" screen so user is not overwhelmed with choices (one for copy myself, one for email/message/from)

File: docassemble/AssemblyLine/data/questions/al_visual.yml [hunk 1]
```diff
@@ -66,3 +78,224 @@ image sets:
...
+---
+template: al_tell_a_friend_message_template
+subject: |
+  ${ al_share_form_from_name } wants to tell you about Court Forms Online
+content: |
+  ${ tell_a_friend_message }
+  ${ interview_url(i=user_info().filename, style="short", new_session=1) }
+---
+objects:
+  - github_readonly: DAWeb.using(base_url="https://api.github.com")
```
Comment (BryceStevenWilley) [inline docassemble/AssemblyLine/data/questions/al_visual.yml:254]: Something to keep in mind: the rate limit for unauthenticated github api calls is 60 an hour. We could easily go over that in prod. Should probably get the github auth token and pass it along. In curl, it's something like should just be the `Authorization: <base64encoded "user:password" string>` header, not sure what DAWeb has.

**Review by BryceStevenWilley — COMMENTED** [review]
LGTM, just the small standup suggestions, and the github api authorization

File: docassemble/AssemblyLine/data/questions/al_visual.yml [hunk 2]
```diff
@@ -40,23 +40,35 @@ code: |
...
     github_repo_name = "docassemble-AssemblyLine"
 ---
+code: |
+  # Username or organization name for Github repositories
+  # Used to direct feedback to the right place and to check for
+  # the last commit date.
+  github_user = "suffolklitlab" 
+---
+code: |
+  feedback_form = "docassemble.MAVirtualCourt:feedback.yml"
```
[the remark this answers is not in the document]
Comment (nonprofittechy) [reply docassemble/AssemblyLine/data/questions/al_visual.yml:61]: Yes, see #49

File: docassemble/AssemblyLine/data/questions/al_visual.yml (hunk 1 above)
Comment (nonprofittechy) [reply docassemble/AssemblyLine/data/questions/al_visual.yml:254]: We already have a token for the feedback form so I will change this API call to make use of it.

**Comment by nonprofittechy** [comment]
Thanks. I will incorporate the rest of the feedback from Friday too before merging.

File: docassemble/AssemblyLine/data/static/styles.css [hunk 3]
```diff
@@ -3,6 +3,14 @@
...
 */
 
+footer {
+  text-align: right
+}
+
+footer a {
+  padding-left: 2em;
+}
+
```
[the remark this answers is not in the document]
Comment (nonprofittechy) [reply docassemble/AssemblyLine/data/static/styles.css:13]: I'm going to be cautious about changing sizing in this PR from past experience--some widths and height changes seem to break docassemble features without a lot of careful related changes. But this is a good note for a future issue.

File: docassemble/AssemblyLine/data/questions/al_visual.yml [hunk 4]
```diff
@@ -66,3 +78,224 @@ image sets:
...
+  else:
+    package_updated_on = None
+---
+event: about_this_interview
+id: about AssemblyLine project
+question: |
+  About this form
+# TODO: I think people might want more form-specific info here?
+subquestion: |
+  [CENTER] [FILE seal.jpg, 30%, Logo of the Suffolk LIT Lab]
```
[the remark this answers is not in the document]
Comment (nonprofittechy) [reply docassemble/AssemblyLine/data/questions/al_visual.yml:272]: I'll check to confirm

File: docassemble/AssemblyLine/data/static/styles.css (hunk 3 above)
[the remark this answers is not in the document]
Comment (BryceStevenWilley) [reply docassemble/AssemblyLine/data/static/styles.css:13]: That's fair, it's a bit out of scope.

**Pull request merged**